src/main.cpp

```diff
@@ -6420,6 +6420,9 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
                 SendExpeditedBlock(*pblock, pfrom);
         }
 
+        CNodeState* state = State(pfrom->GetId());
+        if (state) state->nSyncStartTime = GetTime();  // reset the time because more headers needed
+
```

Comment:
state still requires cs_main lock, also you should put in a DbgAssert as we do for all other state lookups... so 
{
    LOCK(cs_main);
    CNodeState *state = State(nodeid);
    DbgAssert(state != nullptr, return );
    if (state) state->nSyncStartTime = GetTime();
}